privval/signer_requestHandler.go

```diff
@@ -0,0 +1,44 @@
+package privval
+
+import (
+	"fmt"
+
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/types"
+)
+
+func DefaultValidationRequestHandler(privVal types.PrivValidator, req RemoteSignerMsg, chainID string) (RemoteSignerMsg, error) {
+	var res RemoteSignerMsg
+	var err error
+
+	switch r := req.(type) {
+	case *PubKeyRequest:
+		var p crypto.PubKey
+		p = privVal.GetPubKey()
+		res = &PubKeyResponse{p, nil}
+
+	case *SignVoteRequest:
+		err = privVal.SignVote(chainID, r.Vote)
+		if err != nil {
+			res = &SignedVoteResponse{nil, &RemoteSignerError{0, err.Error()}}
+		} else {
+			res = &SignedVoteResponse{r.Vote, nil}
+		}
+
+	case *SignProposalRequest:
+		err = privVal.SignProposal(chainID, r.Proposal)
+		if err != nil {
+			res = &SignedProposalResponse{nil, &RemoteSignerError{0, err.Error()}}
+		} else {
+			res = &SignedProposalResponse{r.Proposal, nil}
+		}
+
+	case *PingRequest:
```

Comment:
I thought we've wanted to remove the pings/pongs

Reply:
Among other things, we wont to avoid multiplexing issues we had with heartbeats in the original implementation. After working on this in the last few days, there were situations where after a disconnect, the listening socket was not able to detect the failure. It was very common to have a sign request being lost. Retrying was not ideal, as I prefer the KMS not to receive the same message twice.
Pings are now much simpler and are not able to interfere with the normal communication. We can discuss it online and I guide you through the implementation if you find that useful.